src/calculus/sum.jl

```diff
@@ -0,0 +1,60 @@
+export Sum
+
+"""
+**Sum of functions**
+
+    Sum(f₁,…,fₖ)
+
+Given functions `f₁` to `fₖ`, returns their sum
+
+```math
+g(x) = ∑_\{i=1\}^k f_i(x).
+```
+
+The functions `fᵢ` should all have the same domain, which will be the domain of
+the result. If all `fᵢ` have gradients then this function will also have a
+gradient.
+
+"""
+
+immutable Sum{T <: Tuple} <: ProximableFunction fs::T end
+
+Sum(fs::Vararg{ProximableFunction}) = Sum((fs...))
+
+# note: is_prox_accurate false because prox in general doesn't exist?
+is_prox_accurate(f::Sum) = false
+is_convex(f::Sum) = all(is_convex.(f.fs))
+is_set(f::Sum) = all(is_set.(f.fs))
+is_singleton(f::Sum) = all(is_singleton.(f.fs))
```

Comment:
This I would avoid defining

Reply:
I mean: the sum of set indicators is the indicator of the intersection; the intersection of singletons is a singleton if the sets are all equal, which is a nasty condition to check I guess, so I would leave this undefined (and let it default to False).